File: list1/GuesstheNumber_java/app/src/main/java/com/example/guessthenumber/MainActivity.java
package com.example.guessthenumber;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Typeface;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Random;

public class MainActivity extends AppCompatActivity {
    private int number;
    private int score;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        setRandomNumber();
    }

    private void setRandomNumber() {
        Random rand = new Random();
        this.number = rand.nextInt(100) + 1;
    }

    public void guess(View view) {
        EditText input = (EditText)findViewById(R.id.input);

        try {
            int guessNumber = Integer.parseInt(input.getText().toString());
            this.score++;
            analyzeInputNumber(guessNumber);
        } catch (NumberFormatException e) {
            showToast("Give number");
        }
    }

    private void analyzeInputNumber(int guessNumber) {
        if (guessNumber > this.number) {
            UpdateScoreText(Typeface.DEFAULT);
            showToast("Too high");
        } else if (guessNumber < this.number) {
            UpdateScoreText(Typeface.DEFAULT);
            showToast("Too low");
        } else {
            UpdateScoreText(Typeface.DEFAULT_BOLD);
            showToast("You guessed!");
            setGuessButtonListener(null);
        }
    }

    private void showToast(String info) {
        Toast.makeText(this, info, Toast.LENGTH_SHORT).show();
    }

    private void setGuessButtonListener(View.OnClickListener fun) {
        Button guessButton = findViewById(R.id.guessButton);
        guessButton.setOnClickListener(fun);
    }

    private void UpdateScoreText(Typeface type) {
        TextView textScore = (TextView) findViewById(R.id.textScore);
        textScore.setText("Score : " + Integer.toString(this.score));
        textScore.setTypeface(type);
    }

    public void restart(View view) {
        this.score = 0;
        UpdateScoreText(Typeface.DEFAULT);
        cleanInput();
        setGuessButtonListener(this::guess);
        setRandomNumber();
    }

    private void cleanInput() {
        EditText input = (EditText)findViewById(R.id.input);
        input.setText("");
    }
}

File: list1/GuessTheNumber_kotlin/app/src/main/java/com/example/guessthenumber/MainActivity.kt
package com.example.guessthenumber

import android.graphics.Typeface
import android.os.Bundle
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.guessthenumber.databinding.ActivityMainBinding
import kotlin.random.Random

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    private var number = 0
    private var score = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        this.binding = ActivityMainBinding.inflate(layoutInflater)
        val view = this.binding.root
        setContentView(view)
        setRandomNumber()
    }

    private fun setRandomNumber() {
        this.number = Random.nextInt(100) + 1
    }

    fun guess(view: View) {
        try {
            val guessNumber = this.binding.input.text.toString().toInt()
            this.score++
            analyzeInputNumber(guessNumber)
        } catch (e: NumberFormatException) {
            showToast("Give number")
        }
    }

    private fun analyzeInputNumber(guessNumber: Int) {
        if (guessNumber > this.number) {
            UpdateScoreText()
            showToast("Too high")
        } else if (guessNumber < this.number) {
            UpdateScoreText()
            showToast("Too low")
        } else {
            UpdateScoreText(Typeface.DEFAULT_BOLD)
            showToast("You guessed!")
            setGuessButtonListener(null)
        }
    }

    private fun showToast(info: String) {
        Toast.makeText(this, info, Toast.LENGTH_SHORT).show()
    }

    private fun setGuessButtonListener(listener: View.OnClickListener?) {
        this.binding.guessButton.setOnClickListener(listener)
    }

    private fun UpdateScoreText(type: Typeface = Typeface.DEFAULT) {
        this.binding.textScore.text = "Score : " + score.toString()
        this.binding.textScore.typeface = type
    }

    fun restart(view: View) {
        this.score = 0
        this.binding.input.setText("")
        UpdateScoreText()
        setGuessButtonListener(this::guess)
        setRandomNumber()
    }
}
